backend/src/user-queries/query-parsing/parse-aggregate.ts

```diff
@@ -0,0 +1,68 @@
+import { AggregateStep, Table } from "@/definitions";
+import { compileTemplate } from "./utils";
+import * as _ from "lodash";
+import * as assert from "assert";
+
+const aggregateTemplateGrouped = `
+let {{ stepName }} = (
+  from {{ from }}
+  group { {{ group }} } (
+    aggregate {
+      {{ as }} = {{ operation }} {{ column }}
+    }
+  )
+)`;
+
+const aggregateTemplateUngrouped = `
+let {{ stepName }} = (
+  from {{ from }}
+  aggregate {
+    {{ as }} = {{ operation }} {{ column }}
+  }
+)`;
+
+export function parseAggregate(
+  aggregateStep: AggregateStep,
+  index: number,
+  tables: Table[],
+): string {
+  // Find the table for the column that the aggregate is being performed on
+  const aggregateTable = _.find(
+    tables,
+    (table) => table.id === aggregateStep.column.table,
+  );
+  assert(aggregateTable, `Table not found: ${aggregateStep.column.table}`);
+
+  let aggregatePrql = "";
+  if (aggregateStep.group.length === 0) {
+    // Aggregate over the whole table
```

Comment:
Nice comment, much clearer what this does, as I'd have to really think through what "group.length" means in this context